File: apps/model-inference-spark/src/main/scala/com/intel/analytics/zoo/apps/model/inference/spark/TextProcessor.scala
package com.intel.analytics.zoo.apps.model.inference.spark

import java.io.File

import com.intel.analytics.zoo.apps.textclassfication.processing.TextProcessing
import com.intel.analytics.zoo.pipeline.inference.InferenceSupportive
import com.intel.analytics.zoo.pipeline.inference.JTensor

class TextProcessor (val stopWordsCount: Int, val sequenceLength: Int, val embeddingFilePath: String) extends TextProcessing  with InferenceSupportive with Serializable {
  val wordToIndexMap = doLoadWordToIndexMap(new File(embeddingFilePath))

  def preprocess(text: String): JTensor = {
    val tensor = doPreprocess(text, stopWordsCount, sequenceLength, wordToIndexMap)
    transferTensorToJTensor(tensor)
  }
}

File: apps/model-inference-examples/text-classification-inference/src/main/java/com/intel/analytics/zoo/apps/textclassfication/inference/TextProcessor.java
package com.intel.analytics.zoo.apps.textclassfication.inference;

import com.intel.analytics.zoo.apps.textclassfication.processing.JTextProcessing;
import com.intel.analytics.zoo.pipeline.inference.JTensor;

import java.io.File;
import java.util.Map;

public class TextProcessor extends JTextProcessing {
    private int stopWordsCount;
    private int sequenceLength;
    private Map<String, Integer> wordToIndexMap;

    public TextProcessor(int stopWordsCount, int sequenceLength, String embeddingFilePath) {
        this.stopWordsCount = stopWordsCount;
        this.sequenceLength = sequenceLength;
        this.wordToIndexMap = loadWordToIndexMap(new File(embeddingFilePath));
    }

    public JTensor preprocess(String text) {
        return preprocess(text, this.stopWordsCount, this.sequenceLength, this.wordToIndexMap);
    }
}
